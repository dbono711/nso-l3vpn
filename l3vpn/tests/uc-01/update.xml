    
<l3vpn xmlns="http://dbono711/l3vpn">
    <customer-name>Disneyland</customer-name>
    <service-id>service01</service-id>
    <vpn-id>1</vpn-id>
    <inet>IPv4-IPv6</inet>
    <max-routes>500</max-routes>
    <max-routes-warning>75</max-routes-warning>
    <provider-edge>
        <asn>65000</asn>
        <device>
            <name>iosxr-0</name>
            <redistribute>static</redistribute>
            <redistribute>connected</redistribute>
            <interface>
                <name>GigabitEthernet2/0</name>
                <cir>100</cir>
                <mtu>2000</mtu>
                <port-mode>true</port-mode>
                <ipv4-local-prefix>10.100.10.1/29</ipv4-local-prefix>
            </interface>
            <interface>
                <name>TenGigE2/0</name>
                <cir>100</cir>
                <mtu>2000</mtu>
                <port-mode>true</port-mode>
                <ipv4-local-prefix>10.100.10.25/29</ipv4-local-prefix>
            </interface>
            <ce-routing>
                <static>
                    <ipv4-destination-prefix>
                        <ipv4-prefix>172.16.16.0/24</ipv4-prefix>
                        <ipv4-forwarding>10.100.10.2</ipv4-forwarding>
                    </ipv4-destination-prefix>
                    <ipv4-destination-prefix>
                        <ipv4-prefix>172.16.17.0/24</ipv4-prefix>
                        <ipv4-forwarding>10.100.10.3</ipv4-forwarding>
                    </ipv4-destination-prefix>
                    <ipv4-destination-prefix>
                        <ipv4-prefix>172.16.19.0/24</ipv4-prefix>
                        <ipv4-forwarding>10.100.10.26</ipv4-forwarding>
                    </ipv4-destination-prefix>
                    <ipv4-destination-prefix>
                        <ipv4-prefix>172.16.20.0/24</ipv4-prefix>
                        <ipv4-forwarding>10.100.10.27</ipv4-forwarding>
                    </ipv4-destination-prefix>
                </static>
            </ce-routing>
        </device>
        <device>
            <name>iosxr-1</name>
            <redistribute>static</redistribute>
            <redistribute>connected</redistribute>
            <interface>
                <name>GigabitEthernet2/0</name>
                <cir>100</cir>
                <mtu>2000</mtu>
                <port-mode>false</port-mode>
                <efp-id>1</efp-id>
                <vlan-id>21</vlan-id>
                <ipv4-local-prefix>10.100.10.9/29</ipv4-local-prefix>
            </interface>
            <interface>
                <name>TenGigE2/0</name>
                <cir>100</cir>
                <mtu>2000</mtu>
                <port-mode>false</port-mode>
                <efp-id>1</efp-id>
                <vlan-id>21</vlan-id>
                <vlan-id>121</vlan-id>
                <ipv6-local-prefix>c0da::1/64</ipv6-local-prefix>
            </interface>
            <ce-routing>
                <static>
                    <ipv4-destination-prefix>
                        <ipv4-prefix>172.16.18.0/24</ipv4-prefix>
                        <ipv4-forwarding>10.100.10.10</ipv4-forwarding>
                    </ipv4-destination-prefix>
                    <ipv6-destination-prefix>
                        <ipv6-prefix>abcd::/64</ipv6-prefix>
                        <ipv6-forwarding>c0da::2</ipv6-forwarding>
                        <ipv6-forwarding>c0da::3</ipv6-forwarding>
                    </ipv6-destination-prefix>
                </static>
            </ce-routing>
        </device>
        <device>
            <name>iosxr-2</name>
            <redistribute>static</redistribute>
            <redistribute>connected</redistribute>
            <interface>
                <name>GigabitEthernet2/0</name>
                <cir>100</cir>
                <mtu>2000</mtu>
                <port-mode>false</port-mode>
                <efp-id>1</efp-id>
                <vlan-id>21</vlan-id>
                <ipv4-local-prefix>10.100.10.17/29</ipv4-local-prefix>
            </interface>
            <interface>
                <name>GigabitEthernet2/1</name>
                <cir>100</cir>
                <mtu>2000</mtu>
                <port-mode>false</port-mode>
                <efp-id>2</efp-id>
                <vlan-id>22</vlan-id>
                <vlan-id>122</vlan-id>
                <ipv6-local-prefix>c0db::1/64</ipv6-local-prefix>
            </interface>
            <ce-routing>
                <static>
                    <ipv6-destination-prefix>
                        <ipv6-prefix>abc9::/64</ipv6-prefix>
                        <ipv6-forwarding>c0db::2</ipv6-forwarding>
                    </ipv6-destination-prefix>
                </static>
                <bgp>
                    <asn>65001</asn>
                    <ipv4-neighbor>
                        <address>10.100.10.18</address>
                        <password>$8$b2LmdEaWuM3vrqCL/plZ0oVMmUoFRl9h56ORBQdf8vg=</password>
                    </ipv4-neighbor>
                </bgp>
            </ce-routing>
        </device>
    </provider-edge>
</l3vpn>