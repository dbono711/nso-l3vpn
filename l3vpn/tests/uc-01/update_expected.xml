<dry-run-result xmlns='http://tail-f.com/ns/rest/dry-run'>
    <result-xml>
        <local-node>
            <data>
                <devices xmlns="http://tail-f.com/ns/ncs">
                    <device>
                        <name>iosxr-0</name>
                        <config>
                            <vrf xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <vrf-list>
                                    <name>service01:1</name>
                                    <address-family>
                                        <ipv6>
                                            <unicast>
                                                <import>
                                                    <route-target>
                                                        <address-list>
                                                            <name>65000:1</name>
                                                        </address-list>
                                                    </route-target>
                                                </import>
                                                <export>
                                                    <route-target>
                                                        <address-list>
                                                            <name>65000:1</name>
                                                        </address-list>
                                                    </route-target>
                                                </export>
                                                <maximum>
                                                    <prefix>
                                                        <limit>500</limit>
                                                        <mid-thresh>75</mid-thresh>
                                                    </prefix>
                                                </maximum>
                                            </unicast>
                                        </ipv6>
                                    </address-family>
                                </vrf-list>
                            </vrf>
                            <interface xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <TenGigE>
                                    <id>2/0</id>
                                    <description>NSO:Disneyland:service01</description>
                                    <mtu>2000</mtu>
                                    <service-policy>
                                        <input>
                                            <name>service01</name>
                                        </input>
                                    </service-policy>
                                    <vrf>service01:1</vrf>
                                    <ipv4>
                                        <address>
                                            <ip>10.100.10.25</ip>
                                            <mask>255.255.255.248</mask>
                                        </address>
                                    </ipv4>
                                </TenGigE>
                            </interface>
                            <router xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <static>
                                    <vrf>
                                        <name>service01</name>
                                        <address-family>
                                            <ipv4>
                                                <unicast>
                                                    <routes-ip>
                                                        <net>172.16.19.0/24</net>
                                                        <address>10.100.10.26</address>
                                                        <description>service01</description>
                                                    </routes-ip>
                                                    <routes-ip>
                                                        <net>172.16.20.0/24</net>
                                                        <address>10.100.10.27</address>
                                                        <description>service01</description>
                                                    </routes-ip>
                                                </unicast>
                                            </ipv4>
                                        </address-family>
                                    </vrf>
                                </static>
                                <bgp>
                                    <bgp-no-instance>
                                        <id>65000</id>
                                        <vrf>
                                            <name>service01:1</name>
                                            <address-family>
                                                <ipv6>
                                                    <unicast>
                                                        <redistribute>
                                                            <connected></connected>
                                                            <static></static>
                                                        </redistribute>
                                                    </unicast>
                                                </ipv6>
                                            </address-family>
                                        </vrf>
                                    </bgp-no-instance>
                                </bgp>
                            </router>
                        </config>
                    </device>
                    <device>
                        <name>iosxr-1</name>
                        <config>
                            <vrf xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <vrf-list>
                                    <name>service01:1</name>
                                    <address-family>
                                        <ipv6>
                                            <unicast>
                                                <import>
                                                    <route-target>
                                                        <address-list>
                                                            <name>65000:1</name>
                                                        </address-list>
                                                    </route-target>
                                                </import>
                                                <export>
                                                    <route-target>
                                                        <address-list>
                                                            <name>65000:1</name>
                                                        </address-list>
                                                    </route-target>
                                                </export>
                                                <maximum>
                                                    <prefix>
                                                        <limit>500</limit>
                                                        <mid-thresh>75</mid-thresh>
                                                    </prefix>
                                                </maximum>
                                            </unicast>
                                        </ipv6>
                                    </address-family>
                                </vrf-list>
                            </vrf>
                            <interface xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <TenGigE-subinterface>
                                    <TenGigE>
                                        <id>2/0.1</id>
                                        <description>NSO:Disneyland:service01</description>
                                        <mtu>2000</mtu>
                                        <encapsulation>
                                            <dot1q>
                                                <vlan-id>21</vlan-id>
                                                <vlan-id>121</vlan-id>
                                            </dot1q>
                                        </encapsulation>
                                        <service-policy>
                                            <input>
                                                <name>service01</name>
                                            </input>
                                        </service-policy>
                                        <vrf>service01:1</vrf>
                                        <ipv6>
                                            <address>
                                                <prefix-list>
                                                    <prefix>c0da::1/64</prefix>
                                                </prefix-list>
                                            </address>
                                        </ipv6>
                                    </TenGigE>
                                </TenGigE-subinterface>
                            </interface>
                            <router xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <static>
                                    <vrf>
                                        <name>service01</name>
                                        <address-family>
                                            <ipv6>
                                                <unicast>
                                                    <routes-ip>
                                                        <net>abcd::/64</net>
                                                        <address>c0da::2</address>
                                                        <description>service01</description>
                                                    </routes-ip>
                                                    <routes-ip>
                                                        <net>abcd::/64</net>
                                                        <address>c0da::3</address>
                                                        <description>service01</description>
                                                    </routes-ip>
                                                </unicast>
                                            </ipv6>
                                        </address-family>
                                    </vrf>
                                </static>
                                <bgp>
                                    <bgp-no-instance>
                                        <id>65000</id>
                                        <vrf>
                                            <name>service01:1</name>
                                            <address-family>
                                                <ipv6>
                                                    <unicast>
                                                        <redistribute>
                                                            <connected></connected>
                                                            <static></static>
                                                        </redistribute>
                                                    </unicast>
                                                </ipv6>
                                            </address-family>
                                        </vrf>
                                    </bgp-no-instance>
                                </bgp>
                            </router>
                        </config>
                    </device>
                    <device>
                        <name>iosxr-2</name>
                        <config>
                            <vrf xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <vrf-list>
                                    <name>service01:1</name>
                                    <address-family>
                                        <ipv6>
                                            <unicast>
                                                <import>
                                                    <route-target>
                                                        <address-list>
                                                            <name>65000:1</name>
                                                        </address-list>
                                                    </route-target>
                                                </import>
                                                <export>
                                                    <route-target>
                                                        <address-list>
                                                            <name>65000:1</name>
                                                        </address-list>
                                                    </route-target>
                                                </export>
                                                <maximum>
                                                    <prefix>
                                                        <limit>500</limit>
                                                        <mid-thresh>75</mid-thresh>
                                                    </prefix>
                                                </maximum>
                                            </unicast>
                                        </ipv6>
                                    </address-family>
                                </vrf-list>
                            </vrf>
                            <interface xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <GigabitEthernet-subinterface>
                                    <GigabitEthernet>
                                        <id>2/1.2</id>
                                        <description>NSO:Disneyland:service01</description>
                                        <mtu>2000</mtu>
                                        <encapsulation>
                                            <dot1q>
                                                <vlan-id>22</vlan-id>
                                                <vlan-id>122</vlan-id>
                                            </dot1q>
                                        </encapsulation>
                                        <service-policy>
                                            <input>
                                                <name>service01</name>
                                            </input>
                                        </service-policy>
                                        <vrf>service01:1</vrf>
                                        <ipv6>
                                            <address>
                                                <prefix-list>
                                                    <prefix>c0db::1/64</prefix>
                                                </prefix-list>
                                            </address>
                                        </ipv6>
                                    </GigabitEthernet>
                                </GigabitEthernet-subinterface>
                            </interface>
                            <router xmlns="http://tail-f.com/ned/cisco-ios-xr">
                                <static>
                                    <vrf>
                                        <name>service01</name>
                                        <address-family>
                                            <ipv6>
                                                <unicast>
                                                    <routes-ip>
                                                        <net>abc9::/64</net>
                                                        <address>c0db::2</address>
                                                        <description>service01</description>
                                                    </routes-ip>
                                                </unicast>
                                            </ipv6>
                                        </address-family>
                                    </vrf>
                                </static>
                                <bgp>
                                    <bgp-no-instance>
                                        <id>65000</id>
                                        <neighbor-group>
                                            <name>service01</name>
                                            <address-family>
                                                <ipv6>
                                                    <unicast>
                                                        <route-policy>
                                                            <direction>in</direction>
                                                            <name>PASS</name>
                                                        </route-policy>
                                                        <route-policy>
                                                            <direction>out</direction>
                                                            <name>PASS</name>
                                                        </route-policy>
                                                        <soft-reconfiguration>
                                                            <inbound>
                                                                <type>always</type>
                                                            </inbound>
                                                        </soft-reconfiguration>
                                                    </unicast>
                                                </ipv6>
                                            </address-family>
                                        </neighbor-group>
                                        <vrf>
                                            <name>service01:1</name>
                                            <address-family>
                                                <ipv4>
                                                    <unicast>
                                                        <redistribute>
                                                            <static></static>
                                                        </redistribute>
                                                    </unicast>
                                                </ipv4>
                                                <ipv6>
                                                    <unicast>
                                                        <redistribute>
                                                            <connected></connected>
                                                            <static></static>
                                                        </redistribute>
                                                    </unicast>
                                                </ipv6>
                                            </address-family>
                                        </vrf>
                                    </bgp-no-instance>
                                </bgp>
                            </router>
                        </config>
                    </device>
                </devices>
                <services xmlns="http://tail-f.com/ns/ncs">
                    <l3vpn xmlns="http://dbono711/l3vpn">
                        <customer-name>Disneyland</customer-name>
                        <service-id>service01</service-id>
                        <inet>IPv4-IPv6</inet>
                        <provider-edge>
                            <device>
                                <name>iosxr-0</name>
                                <interface>
                                    <name>TenGigE2/0</name>
                                    <cir>100</cir>
                                    <mtu>2000</mtu>
                                    <port-mode>true</port-mode>
                                    <ipv4-local-prefix>10.100.10.25/29</ipv4-local-prefix>
                                </interface>
                                <ce-routing>
                                    <static>
                                        <ipv4-destination-prefix>
                                            <ipv4-prefix>172.16.19.0/24</ipv4-prefix>
                                            <ipv4-forwarding>10.100.10.26</ipv4-forwarding>
                                        </ipv4-destination-prefix>
                                        <ipv4-destination-prefix>
                                            <ipv4-prefix>172.16.20.0/24</ipv4-prefix>
                                            <ipv4-forwarding>10.100.10.27</ipv4-forwarding>
                                        </ipv4-destination-prefix>
                                    </static>
                                </ce-routing>
                            </device>
                            <device>
                                <name>iosxr-1</name>
                                <interface>
                                    <name>TenGigE2/0</name>
                                    <cir>100</cir>
                                    <mtu>2000</mtu>
                                    <port-mode>false</port-mode>
                                    <efp-id>1</efp-id>
                                    <vlan-id>21</vlan-id>
                                    <vlan-id>121</vlan-id>
                                    <ipv6-local-prefix>c0da::1/64</ipv6-local-prefix>
                                </interface>
                                <ce-routing>
                                    <static>
                                        <ipv6-destination-prefix>
                                            <ipv6-prefix>abcd::/64</ipv6-prefix>
                                            <ipv6-forwarding>c0da::2</ipv6-forwarding>
                                            <ipv6-forwarding>c0da::3</ipv6-forwarding>
                                        </ipv6-destination-prefix>
                                    </static>
                                </ce-routing>
                            </device>
                            <device>
                                <name>iosxr-2</name>
                                <redistribute>static</redistribute>
                                <interface>
                                    <name>GigabitEthernet2/1</name>
                                    <cir>100</cir>
                                    <mtu>2000</mtu>
                                    <port-mode>false</port-mode>
                                    <efp-id>2</efp-id>
                                    <vlan-id>22</vlan-id>
                                    <vlan-id>122</vlan-id>
                                    <ipv6-local-prefix>c0db::1/64</ipv6-local-prefix>
                                </interface>
                                <ce-routing>
                                    <static>
                                        <ipv6-destination-prefix>
                                            <ipv6-prefix>abc9::/64</ipv6-prefix>
                                            <ipv6-forwarding>c0db::2</ipv6-forwarding>
                                        </ipv6-destination-prefix>
                                    </static>
                                </ce-routing>
                            </device>
                        </provider-edge>
                    </l3vpn>
                </services>
            </data>
        </local-node>
    </result-xml>
</dry-run-result>